app/javascript/packages/analytics/click-observer-element.ts

```diff
@@ -0,0 +1,45 @@
+import { trackEvent } from '.';
+
+class ClickObserverElement extends HTMLElement {
+  trackEvent: typeof trackEvent = trackEvent;
+
+  connectedCallback() {
+    this.addEventListener('click', (event) => this.handleEvent(event), true);
+    this.addEventListener('change', (event) => this.handleEvent(event), true);
+  }
+
+  get eventName(): string | null {
+    return this.getAttribute('event-name');
+  }
+
+  /**
+   * Whether event handling should handle target as a checkbox.
+   */
+  get isHandledAsCheckbox(): boolean {
+    return !!this.querySelector('[type=checkbox]');
+  }
+
+  handleEvent(event: Event) {
+    if (!this.eventName) {
+      return;
+    }
+
+    if (event.type === 'change' && this.isHandledAsCheckbox) {
+      this.trackEvent(this.eventName, { checked: (event.target as HTMLInputElement).checked });
+    } else if (event.type === 'click' && !this.isHandledAsCheckbox) {
+      this.trackEvent(this.eventName);
+    }
```

Comment:
I switched up the implementation here to more explicitly handle checkboxes, since I discovered that it's possible for two click events to be emitted when clicking on the label associated with a checkbox, which would have been an unfortunate bug to have gone live.

I was hoping to handle it differently by deduping events, but it appeared that the events were distinct instances with no way I could see to compare them.

Previous attempt was something like...

```ts
const handledEvents = new WeakMap<Event, boolean>();

// ...

handleClick(event: MouseEvent) {
  if (handledEvents.has(event)) {
    return;
  }

  handledEvents.set(event, true);

  // ...
}

```